bundles/org.openhab.binding.govee/src/main/resources/OH-INF/config/config.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<config-description:config-descriptions
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:config-description="https://openhab.org/schemas/config-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/config-description/v1.0.0 https://openhab.org/schemas/config-description-1.0.0.xsd">
+
+	<config-description uri="thing-type:govee:govee-light">
+		<parameter name="hostname" type="text" required="true">
+			<context>network-address</context>
+			<label>Hostname/IP Address</label>
+			<description>Hostname or IP address of the device</description>
+		</parameter>
+		<parameter name="macAddress" type="text" required="true">
+			<label>MAC Address</label>
+			<description>MAC Address of the device</description>
+		</parameter>
+		<parameter name="deviceType" type="text" required="true">
+			<label>Device Product Number (sku)</label>
+			<description>The product number of the device</description>
+		</parameter>
+		<parameter name="productName" type="text" required="false">
+			<label>Device Product Name</label>
+			<description>Description of the device</description>
+		</parameter>
```

Comment:
I don't understand what you mean with properties. If you create a thing, these should be definitely provided even though the device could work without it but still I feel they should be editable in case the user wants to overwrite them.

Reply:
Why would the user need to edit the product name? The binding doesn"t use it for anything, I believe?

Reply:
The binding does not need it but the to me it is important it is shown.
If the product name is discovered incorrectly the user can overwrite it. I wouldn't say that the binding can be 100% correct in detecting it. It only discovers the SKU but the product name is best guessing.

Reply:
I can understand @jlaur though. If the binding doesn't actually use the product name, and it neither can be meaningfully discovered, what's the point in showing it in the first place? If it can be discovered properly, it should be a property; if the binding needs it and it can't be discovered properly, it should be a parameter, but if neither applies, I think it shouldn't exist at all.
Don't forget you'll make users think about what to achieve when editing this value. They'll be confused when they hear the answer is 'nothing' ;-)

Reply:
@maniac103 The device CAN discover it, it receives the the SKU and IF it knows this SKU it automatically even provides a name for it. BUT IF the SKU is unknown for a device that is still working (which happened just the other week because a device that is not in the official list actually works), the binding will still still put that sku into deviceType but will fill the productName with "???".

So, 
- macAddress is mandatory for the device to work (as it builds the uid)
- hostname is mandatory for the device to work (to communicate)
- deviceType will always be discovered, so you can argue it doesn't need to be edited afterwards (readOnly)
- productName should be editable as the user might want to overwrite of the description that the binding provides (editable)
- wifiHardwareVersion can be discovered and should be shown to the user (readOnly)
- wifiSoftwareVersion  can be discovered and should be shown to the user (readOnly)

So how exactly with which tag and which element do I define these values?
- When do I use <parameter?
- When do I use <properties>?
- When do I use <representation-property>
(what does it mean below "you don't do declare these?"

Reply:
`parameter` is meant for values the binding can't detect by itself and which are needed for the binding to work.
`property` is meant for read-only values the binding can determine by itself and which are used by the binding itself and/or provided to the user for informational purposes.
`representation-property` is the property type to be used as kind of ID value. The inbox will distinguish discovery results by that value.

So I'd agree with your list, with the exception of productName: as the binding does not use it by itself, I'd say it should simply be omitted in cases where it can't be determined. Same applies to other properties which can only be determined in some cases. After all, there's no additional value provided for the user by showing values with question marks.
(NB: I'm not entirely sure about this, but I think if thing for a given representation property exists already and later discovery runs detect a new property, that property is added to the existing thing).

Reply:
I removed the two hw/sw versions, I made the macAddress a representation property. I made the device type and hostname readonly which looks nice and good after discovery:

<img width="769" alt="image" src="https://github.com/openhab/openhab-addons/assets/5937600/10634dd8-556e-473a-b87a-2efcc7b742ad">


**But here is the catch:** if someone wants to create device manually, then it does't work

<img width="844" alt="image" src="https://github.com/openhab/openhab-addons/assets/5937600/e1d3a90b-2786-4c52-9119-e63e36e67773">

- the macAdress isn't shown at all, so it cannot be entered which is a blocker
- hostname and sku are readonly even here, which is a blocker (at least for the hostname)
- "of course" the properties sw/hw version cannot also not be supplied manually.

Reply:
MAC address needs to be a parameter, same as hostname. Sorry for being ambiguous above, I thought the relation 'is mandatory' -> 'must be parameter' was clear. For discovery results, those parameters are initialized from the values in the respective `withProperty` calls in the `DiscoveryResult`. IOW, the configuration map, which contains the parameter values, is initialized from the property map.

SKU and the versions can probably stay as properties and be updated on thing initialization - similarly to e.g. [here](https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.ecovacs/src/main/java/org/openhab/binding/ecovacs/internal/handler/EcovacsVacuumHandler.java#L520)

Reply:
Well, no, I though about that a while: if I _can and do_ discover it (due to a scan being started) then I get all information but when someone adds the thing manually I won't trigger a discovery for that, so in that case the sku and product would stay empty. 

Maybe this discussion is kind of philosophical in terms of manual creation. The likelihood that someone would actually create it manually is very small. I would even think that manual creating in a thing file is actually pretty hard because you would need to find out the hostname and mac address of that device which is very hard to find out without network knowledge. 
To compare that with hues: Do people ever add hue lights manually and if, how would they know all the parameters about that? 

Which then would lead to the question, how do people deal with that who don't like the WebUI and want to create their things in files?

Reply:
> Is it actually possible in the file definition to provide thing properties?

I don't think so, but I also don't think there's an actual loss here: properties help with identifying / distinguishing discovery results, but when manually creating things, the user already knows what device he's adding. So while not having the properties in the manual case may not be ideal/perfect, I don't think it's an actual issue.

Reply:
> The likelihood that someone would actually create it manually is very small. I would even think that manual creating in a thing file is actually pretty hard because you would need to find out the hostname and mac address of that device which is very hard to find out without network knowledge. To compare that with hues: Do people ever add hue lights manually and if, how would they know all the parameters about that?
> 
> Which then would lead to the question, how do people deal with that who don't like the WebUI and want to create their things in files?

I create **all** my Things manually. Sometimes I need to use discovery first and create a managed Thing. I can then copy the configuration parameters into my manually created Thing in a .things file and delete the managed Thing again.